src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -189,6 +189,13 @@ private sealed class CachedState
                         minimumVSDefinedSDKVersion);
                 }
 
+                string dotnetExe = Path.Combine(dotnetRoot, "dotnet.exe");
```

Comment:
Do we have `Muxer.cs` available from `Microsoft.DotNet.Cli.Utils` at this point? We should use `new Muxer().MuxerPath` for the path to `dotnet.exe`, and we should use `new Muxer().SharedFxVersion` as the value for the `MSBUILD_NET_TASKHOST_RUNTIME_VERSION` property.

Reply:
When I tried to run `dotnet build` to build a console app, the `MSBuildSdkResolver` was not in the list of the resolvers. When I tried to run `msbuild` the `MSBuildSdkResolver` was in the list. I also found this comment in msbuild repo that explains the flow of sdk resolution:
https://github.com/dotnet/msbuild/blob/6ca999210ce0e6541d92532874b732c4fceee8de/src/Build/BackEnd/Components/SdkResolution/SdkResolverService.cs#L113-L138

`MuxerPath` gets the current process's path. In case of `dotnet` it gets the path to `dotnet` that is running.

Reply:
ah, of course - for `dotnet build` there is no MSBuildSdkResolver because the SDK sets up the variables required for the MSBuild DefaultSdkResolver to Just Work.